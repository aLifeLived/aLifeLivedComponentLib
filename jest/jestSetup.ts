/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
