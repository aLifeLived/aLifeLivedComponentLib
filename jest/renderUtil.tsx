import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
