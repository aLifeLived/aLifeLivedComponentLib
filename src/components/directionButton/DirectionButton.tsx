import React from 'react';
import { Box } from '../../themes/box/Box';
import { Text } from '../../themes/text/Text';
import {
  StyledOpacityProps,
  SystemButton,
} from '../../themes/systemButton/SystemButton';
import { Icon } from '../icon/Icon';
import { IconProp } from '@fortawesome/fontawesome-svg-core';

interface IDirectionButtonProps {
  title: string;
  onPress: () => void;
  isDisabled?: boolean;
  variant?: 'cta';
  color?: string;
  leftIconName?: IconProp;
}

export const DirectionButton: React.FC<
  IDirectionButtonProps & StyledOpacityProps
> = ({ title, onPress, isDisabled, color, leftIconName, ...rest }) => {
  return (
    <SystemButton
      onPress={() => onPress()}
      paddingBottom="m"
      paddingTop="m"
      paddingHorizontal="s"
      borderBottomWidth={1}
      borderBottomColor="backgroundContrastMid"
      width="100%"
      disabled={isDisabled}
      {...rest}
    >
      <Box
        flexDirection="row"
        alignItems="center"
        justifyContent="space-between"
      >
        <Box flexDirection="row">
          {leftIconName && (
            <Icon
              icon={leftIconName}
              color="primary"
              size={20}
              marginRight="s"
              testID="direction-button-left-icon"
            />
          )}
          <Text variant="body" textAlign="center" color={color}>
            {title}
          </Text>
        </Box>
        <Box>
          <Icon icon="chevron-right" color="secondary" size={20} />
        </Box>
      </Box>
    </SystemButton>
  );
};
