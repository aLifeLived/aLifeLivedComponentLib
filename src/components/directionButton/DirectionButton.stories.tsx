import React from 'react';

import { storiesOf } from '@storybook/react-native';

// Component
import { DirectionButton } from './DirectionButton';
import { action } from '@storybook/addon-actions';
import { CenterView } from '../centerView/CenterView';
import { Box } from '../../themes/box/Box';

storiesOf('Button', module)
  .addDecorator(Story => <CenterView>{Story()}</CenterView>)
  .add('Direction button', () => (
    <Box width="100%" alignItems="center">
      <Box width="90%" margin="m">
        <Box margin="m">
          <DirectionButton
            title="Account"
            onPress={action('Button on press fired')}
            isDisabled={false}
            leftIconName="lightbulb"
          />
        </Box>
        <Box margin="m">
          <DirectionButton
            title="Billing"
            onPress={action('Button on press fired')}
            isDisabled={false}
            leftIconName="bookmark"
          />
        </Box>
        <Box margin="m">
          <DirectionButton
            title="Stories"
            onPress={action('Button on press fired')}
            isDisabled={false}
            leftIconName="play"
          />
        </Box>
      </Box>
    </Box>
  ));
