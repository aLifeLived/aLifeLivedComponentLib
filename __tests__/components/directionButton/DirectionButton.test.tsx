import React from 'react';
import { render, fireEvent } from './../../../jest/renderUtil';

// Component
import { DirectionButton } from '../../../src/components/directionButton/DirectionButton';

const mockOnPress = jest.fn();
const buttonTitle = 'Test button';
const setup = ({ isDisabled = false, leftIconName = undefined }) =>
  render(
    <DirectionButton
      title={buttonTitle}
      onPress={mockOnPress}
      isDisabled={isDisabled}
      leftIconName={leftIconName}
    />
  );

describe('components >> directionButton ', () => {
  describe('Given direction button has a left icon', () => {
    test('Should render direction button correctly', () => {
      const { toJSON } = setup({
        isDisabled: false,
        leftIconName: 'microphone',
      });
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render button with left icon', () => {
      const { getByTestId } = setup({
        isDisabled: false,
        leftIconName: 'microphone',
      });
      expect(getByTestId('direction-button-left-icon')).toBeDefined();
    });
  });

  describe('Given direction button is enabled', () => {
    test('Should render direction button correctly', () => {
      const { toJSON } = setup({ isDisabled: false });
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render title', () => {
      const { getByText } = setup({ isDisabled: false });
      expect(getByText(buttonTitle)).toBeDefined();
    });

    test('Should fire on press', () => {
      const { getByText } = setup({ isDisabled: false });
      fireEvent.press(getByText(buttonTitle));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });

  describe('Given direction button is disabled', () => {
    beforeAll(() => mockOnPress.mockClear());
    test('Should find title', () => {
      const { getByText } = setup({ isDisabled: true });
      expect(getByText(buttonTitle)).toBeDefined();
    });

    test('Should NOT be able to fire on press action', () => {
      const { getByText } = setup({ isDisabled: true });
      fireEvent.press(getByText(buttonTitle));
      expect(mockOnPress).toBeCalledTimes(0);
    });
  });
});
