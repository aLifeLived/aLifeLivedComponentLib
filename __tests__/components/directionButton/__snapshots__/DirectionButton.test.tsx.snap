// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components >> directionButton  Given direction button has a left icon Should render direction button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#CBCBCB",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingHorizontal": 8,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="#67B6A8"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={20}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "marginRight": 8,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        testID="direction-button-left-icon"
        vbHeight={512}
        vbWidth={352}
        width={20}
      >
        <RNSVGGroup
          fill={4284987048}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
            fill={4284987048}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontFamily": "SFUIDisplay-Regular",
              "fontSize": 18,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Test button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="#F08969"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={20}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={20}
      >
        <RNSVGGroup
          fill={4293953897}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill={4293953897}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`components >> directionButton  Given direction button is enabled Should render direction button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#CBCBCB",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 16,
      "paddingHorizontal": 8,
      "paddingTop": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#282828",
              "display": "flex",
              "fontFamily": "SFUIDisplay-Regular",
              "fontSize": 18,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Test button
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "display": "flex",
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        fill="#F08969"
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        secondaryFill={null}
        secondaryOpacity={0.4}
        size={20}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {},
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={512}
        vbWidth={320}
        width={20}
      >
        <RNSVGGroup
          fill={4293953897}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            fill={4293953897}
            fillOpacity={1}
            propList={
              Array [
                "fill",
                "fillOpacity",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
